controller/k8s/api.go

```diff
@@ -424,22 +426,38 @@ func (api *API) GetPodsFor(obj runtime.Object, includeFailed bool) ([]*corev1.Po
 	case *appsv1.DaemonSet:
 		namespace = typed.Namespace
 		selector = labels.Set(typed.Spec.Selector.MatchLabels).AsSelector()
+		uid = typed.UID
 
 	case *appsv1beta2.Deployment:
 		namespace = typed.Namespace
 		selector = labels.Set(typed.Spec.Selector.MatchLabels).AsSelector()
+		ret, err := api.RS().Lister().ReplicaSets(namespace).List(selector)
+		if err != nil {
+			return nil, err
+		}
+		for _, rs := range ret {
+			if rs.Annotations["deployment.kubernetes.io/revision"] == typed.Annotations["deployment.kubernetes.io/revision"] &&
```

Comment:
This will only select the pods from the current revision, correct?  I think we probably want to return all pods managed by the deployment, including all revisions.

Reply:
You're right, I put some effort to just show pods for the same version and completed forgot we want to return pods from all revisions. I changed it to return it properly and added a test case for it.